backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbsLogger.java

```diff
@@ -0,0 +1,84 @@
+package backtraceio.library.breadcrumbs;
+
+import java.io.IOException;
+import java.util.Map;
+
+import backtraceio.library.enums.BacktraceBreadcrumbLevel;
+import backtraceio.library.enums.BacktraceBreadcrumbType;
+import backtraceio.library.logger.BacktraceLogger;
+
+public class BacktraceBreadcrumbsLogger {
+
+    private final String LOG_TAG = BacktraceBreadcrumbsLogger.class.getSimpleName();
+
+    private long breadcrumbId = 0;
+
+    private BacktraceQueueFileHelper backtraceQueueFileHelper;
+
+    /**
+     * We truncate messages longer than this
+     */
+    private final int maxMessageSizeBytes = 1024;
+
+    /**
+     * We truncate serialized attribute strings longer than this
+     */
+    private final int maxSerializedAttributeSizeBytes = 1024;
+
+    public BacktraceBreadcrumbsLogger(String breadcrumbLogDirectory, int maxQueueFileSizeBytes) throws IOException, NoSuchMethodException {
+        this.backtraceQueueFileHelper = new BacktraceQueueFileHelper(breadcrumbLogDirectory, maxQueueFileSizeBytes);
+    }
+
+    public boolean addBreadcrumb(String message, Map<String, Object> attributes, BacktraceBreadcrumbType type, BacktraceBreadcrumbLevel level) {
+        // We use currentTimeMillis in the BacktraceReport too, so for consistency
+        // we will use it here.
+        long time = System.currentTimeMillis();
+
+        message = message.substring(0, Math.min(message.length(), maxMessageSizeBytes));
+
+        StringBuilder breadcrumb = new StringBuilder("\ntimestamp " + Long.toString(time));
+        breadcrumb.append(" id " + Long.toString(breadcrumbId));
```

Comment:
If we're using StringBuilder here - what is the reason of using stringbuilder and string concatenation? I would recommend to avoid string concatenation and use only stringbuilder.
```
StringBuilder breadcrumb = new StringBuilder("\ntimestamp ");
breadcrumbs.append( Long.toString(time));
breadcrumb.append(" id ");....
```

Reply:
Ah that's a good point, I will make this change!